web/src/features/moreCast2/forecastDraft.ts

```diff
@@ -0,0 +1,78 @@
+import { DraftMorecast2Rows, MoreCast2ForecastRow, MoreCast2Row } from 'features/moreCast2/interfaces'
+import { rowContainsActual } from 'features/moreCast2/util'
+import { DateTime } from 'luxon'
+
+export class MorecastDraftForecast {
+  private readonly STORAGE_KEY = 'morecastForecastDraft'
+  private localStorage: Storage
+
+  constructor(localStorage: Storage) {
+    this.localStorage = localStorage
+  }
+
+  public getStoredDraftForecasts = (): DraftMorecast2Rows => {
+    const storedDraftString = this.localStorage.getItem(this.STORAGE_KEY)
+    let storedDraft: DraftMorecast2Rows = { rows: [], lastEdited: 0 }
+
+    if (storedDraftString) {
+      storedDraft = JSON.parse(storedDraftString)
+    }
+
+    return storedDraft
+  }
+
+  public storeDraftForecasts = (forecastDraft: DraftMorecast2Rows) => {
+    this.localStorage.setItem(this.STORAGE_KEY, JSON.stringify(forecastDraft))
+  }
+
+  public clearDraftForecast = () => {
+    this.localStorage.removeItem(this.STORAGE_KEY)
+  }
+
+  public updateStoredDraftForecasts = (rowsToStore: MoreCast2Row[]) => {
+    const storedForecastsToUpdate = this.getStoredDraftForecasts()
+    const storedRowsMap = this.getRowsMap(storedForecastsToUpdate.rows)
+
+    rowsToStore.forEach(row => {
+      storedRowsMap.set(row.id, row)
+    })
+    // we only need to store rows that are 'Forecast' rows
+    storedForecastsToUpdate.rows = Array.from(storedRowsMap.values()).filter(row => {
+      return !rowContainsActual(row)
+    })
+    storedForecastsToUpdate.lastEdited = Date.now()
+
+    this.storeDraftForecasts(storedForecastsToUpdate)
+  }
+
+  public deleteRowsFromStoredDraft = (savedRows: MoreCast2ForecastRow[] | MoreCast2Row[]) => {
+    const localStoredForecast = this.getStoredDraftForecasts()
+    const localStoredRows = this.getRowsMap(localStoredForecast.rows)
+    savedRows.forEach(row => {
+      localStoredRows.delete(row.id)
+    })
+    localStoredForecast.rows = Array.from(localStoredRows.values())
+    localStoredForecast.lastEdited = Date.now()
+    this.storeDraftForecasts(localStoredForecast)
+  }
+
+  public getRowsMap = (morecastRows: MoreCast2Row[]): Map<string, MoreCast2Row> => {
```

Comment:
Could this class import `getRowsMap` from `features/morecas2/utils` instead of declaring its own copy?

Reply:
Thanks, something I missed when juggling things around